src/main/scala/org/ensime/sexp/formats/FamilyFormats.scala

```diff
@@ -14,32 +14,37 @@ import org.ensime.sexp._
  * Boilerplate blocked on https://github.com/milessabin/shapeless/issues/238
  */
 trait FamilyFormats {
-
-  // // unlike case classes, include the type hint in the serialised form
-  // // TODO: get this working as part of a family, rather than just standalone
-  // implicit def singletonFormat[T <: Singleton](
-  //   implicit w: Witness.Aux[T]) = new SexpFormat[T] {
-  //   // TODO: scala names https://github.com/milessabin/shapeless/issues/256
-  //   private val sexp = SexpSymbol(w.value.getClass.getName)
-  //   private val value = w.value
-  //   def write(t: T) = sexp
-  //   def read(v: Sexp) =
-  //     if (v == sexp) value
-  //     else deserializationError(v)
-  // }
-
   case class TypeHint[T](hint: SexpSymbol)
   implicit def typehint[T: TypeTag]: TypeHint[T] =
-    TypeHint(SexpSymbol(":" + typeOf[T].dealias.toString.split("(\\.|\\$)").last))
+    TypeHint(SexpSymbol(":" + typeOf[T].dealias.toString.replaceAll("\\.type$", "").split("(\\.|\\$)").last))
+
+  // always serialises to Nil, and is differentiated by the TraitFormat
+  //
+  // this is not implicit because of a stray println in shapeless that
+  // makes this extremely frustrating to use.
+  // https://github.com/milessabin/shapeless/pull/259
```

Comment:
I tried upgrading to shapeless 2.1 but it broke in a non-trivial way that even miles didn't know the answer to. I think an upgrade will be easier to debug after I modularise (coming next!)